src/Gaffer/VTuneMonitor.cpp

```diff
@@ -0,0 +1,126 @@
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Image Engine Design Inc nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "Gaffer/VTuneMonitor.h"
+
+#ifdef GAFFER_VTUNE
+
+#include <stdio.h>
+
+#include "ittnotify.h"
+#include "Gaffer/Process.h"
+#include "Gaffer/Plug.h"
+#include "Gaffer/Node.h"
+
+struct OpaqueDomain
+{
+	OpaqueDomain(const std::string& domain) : imp ( __itt_domain_create( domain.c_str() ) )
+	{}
+
+	__itt_domain *imp;
+};
+
+namespace
+{
+	const IECore::InternedString hashProcessType( "computeNode:hash" );
+
+	typedef const Gaffer::Node* TaskHandleKey;
+
+	typedef std::map<TaskHandleKey, __itt_string_handle*> TaskHandleMap;
+	struct ThreadLocalData
+	{
+		TaskHandleMap taskHandleMap;
+		bool enabled;
+	};
+
+	tbb::enumerable_thread_specific<ThreadLocalData, tbb::cache_aligned_allocator<ThreadLocalData> > g_threadData;
+}
+
+using namespace Gaffer;
+
+VTuneMonitor::VTuneMonitor(const std::string& domain /*= "Gaffer" */, bool monitorHashProcess /* = false */ )
+: m_monitorHashProcess( monitorHashProcess )
+{
+	m_domain.reset( new OpaqueDomain( domain ) );
+}
+
+VTuneMonitor::~VTuneMonitor()
+{
+}
+
+void VTuneMonitor::processStarted( const Process *process )
+{
+	const Node* node = process->plug()->node();
+	TaskHandleKey key = node;
+	ThreadLocalData& localData = g_threadData.local();
+
+	localData.enabled = true;
+
+	if (!m_monitorHashProcess)
+	{
+		localData.enabled = process->type() != hashProcessType;
+	}
+
+	TaskHandleMap::const_iterator it = localData.taskHandleMap.find( key );
+
+	__itt_string_handle* handle = NULL;
+	if (it != localData.taskHandleMap.end() )
```

Comment:
How beneficial is this map of handles? Is it really quicker to do a TLS lookup and then a map search than it is to just call `__itt_string_handle_create( node->typeName() )` each time we want a handle? It feels like they'll be doing something similar internally anyway, except probably using something faster than a map...